app/src/main/java/com/github/sdp_begreen/begreen/fragments/SendPostFragment.kt

```diff
@@ -189,11 +199,69 @@ class SendPostFragment : Fragment() {
 
                 }
 
-                // Update the user and return
-                lifecycleScope.launch {
-                    updateUser(metadata, user)
-                    returnToCamera()
-                }
+                // Obtain reference to Firebase's '.info/connected' node to monitor connection state.
+                val connectedRef = Firebase.database.getReference(".info/connected")
+
+                // Attach a listener for a single event to check if client is currently connected to Firebase.
+                connectedRef.addListenerForSingleValueEvent(object : ValueEventListener {
+                    override fun onDataChange(snapshot: DataSnapshot) {
+                        // Retrieve connectivity state from snapshot. If data is null, assume not connected.
+                        val connected = snapshot.getValue(Boolean::class.java) ?: false
+
+                        // If connected, proceed with user update and photo sharing operations.
+                        if (connected) {
+                            Log.d("TAG", "connected")
+                            // Launch coroutine to perform potentially long-running operations off the main UI thread.
+                            lifecycleScope.launch {
+                                // Update the user with the new photo and return to the camera.
+                                updateUser(metadata, user)
+                                returnToCamera()
+                            }
+                        // If not connected, store metadata and user information for later use when connectivity is re-established.
+                        } else {
+                            Log.d("TAG", "not connected")
+
+                            // Use TinyDB for local data storage.
+                            val tinyDB = TinyDB(this@SendPostFragment.context)
+
+                            // Retrieve existing metadata and add current one.
+                            val metas = tinyDB.getListObject("metas", TrashPhotoMetadata::class.java)
+                            metas.add(metadata!!)
```

Comment:
Here, if you make `getListObject` return a `List` as suggested, the `add` method won't exist anymore, as the `List` is immutable. Instead, I would suggest you to do something like this: `val newMetas = metas + metadata!!`

Reply:
Change made as proposed, thanks!